### PR TITLE
Add marco to support android compile

### PR DESCRIPTION
**Pull request by LukeZheZhu**
modify the code to support android p compile
@xuguangxin

### CODE REVIEW

File: media_driver/linux/common/os/i915/mos_bufmgr.c [hunk 1]
```diff
@@ -4181,6 +4194,56 @@ int mos_get_slice_mask(uint32_t slice_count)
     return bitmask;
 }
 
+#if defined(MEDIA_EXT)
```
Comment (dvrogozh) [inline media_driver/linux/common/os/i915/mos_bufmgr.c]: This part don't relate to Android at all.

File: media_driver/linux/common/os/i915/mos_bufmgr.c [hunk 2]
```diff
@@ -977,7 +977,7 @@ mos_gem_bo_alloc_internal(struct mos_bufmgr *bufmgr,
     pthread_mutex_unlock(&bufmgr_gem->lock);
 
     if (!alloc_from_cache) {
-        struct drm_i915_gem_create create;
+        struct drm_i915_gem_create_android create;
```
Comment (dvrogozh) [inline media_driver/linux/common/os/i915/mos_bufmgr.c]: Here is a grep on the code:
```
[dvrogozh@dvrscl media-driver]$ fgrep -rsn drm_i915_gem_create_android
[dvrogozh@dvrscl media-driver]$
```
As you see, there is no drm_i915_gem_create_android, thus this code simply will not compile.

File: media_driver/linux/common/os/i915/mos_bufmgr.c [hunk 3]
```diff
@@ -3267,6 +3267,11 @@ do_exec2(struct mos_linux_bo *bo, int used, struct mos_linux_context *ctx,
         i915_execbuffer2_set_context_id(execbuf, ctx->ctx_id);
     execbuf.rsvd2 = 0;
 
+#ifdef ANDROID
+    i915_execbuffer2_set_tag(execbuf, tag);
```
Comment (dvrogozh) [inline media_driver/linux/common/os/i915/mos_bufmgr.c]: Here is a grep for the code:
```
[dvrogozh@dvrscl media-driver]$ fgrep -rsn i915_execbuffer2_set_tag
[dvrogozh@dvrscl media-driver]$
```

As you see i915_execbuffer2_set_tag is entirely missing. This code will not compile.

File: media_driver/linux/gen11/ddi/media_libva_caps_g11.cpp [hunk 4]
```diff
@@ -510,10 +513,9 @@ VAStatus MediaLibvaCapsG11::LoadProfileEntrypoints()
     DDI_CHK_RET(status, "Failed to initialize Caps!");
     status = LoadVp9EncProfileEntrypoints();
     DDI_CHK_RET(status, "Failed to initialize Caps!");
-#ifdef ENABLE_KERNELS
     status = LoadNoneProfileEntrypoints();
```
Comment (dvrogozh) [inline media_driver/linux/gen11/ddi/media_libva_caps_g11.cpp]: Why?!

File: media_driver/linux/common/os/i915/mos_bufmgr.c (hunk 1 above)
Comment (LukeZheZhu) [reply media_driver/linux/common/os/i915/mos_bufmgr.c]: I feel so sorry to push wrong code to the repo. I think this time should be right. Thanks you so much

File: media_driver/linux/gen11/ddi/media_libva_caps_g11.cpp [hunk 5]
```diff
@@ -33,6 +33,9 @@
 #include "media_ddi_encode_const.h"
 #include "media_ddi_decode_const_g11.h"
 #include "media_libva_vp.h"
+#ifdef ANDROID
```
Comment (xuguangxin) [inline media_driver/linux/gen11/ddi/media_libva_caps_g11.cpp]: why gen11 related to android?

**Comment by dvrogozh** [comment]
There is no activity in this PR in a while. I believe this can be closed. Open an issue/new PR if needed, please.

**Pull request closed without merging**